internal/manager/run.go

```diff
@@ -140,7 +141,15 @@ func Run(
 	}
 
 	setupLog.Info("Initializing Dataplane Client")
-	eventRecorder := mgr.GetEventRecorderFor(KongClientEventRecorderComponentName)
+	var eventRecorder record.EventRecorder
+	if c.EmitTranslationEvents {
+		setupLog.Info("emit translation event enabled, create event recorder for " + KongClientEventRecorderComponentName)
+		eventRecorder = mgr.GetEventRecorderFor(KongClientEventRecorderComponentName)
+	} else {
+		setupLog.Info("emit translation event disabled, discard all events")
+		// REVIEW: Use record.FakeRecorder to discard all events that is not primarily defined for this. Should we define a "no-op recorder" somewhere instead?
+		eventRecorder = &record.FakeRecorder{}
```

Comment:
I think using `FakeRecorder` is sufficient and it's thread-safe. Additionally, even if we define a "no-op recorder", we don't need to implement any additional logic for it, right?

Reply:
Yes, I checked the code of `FakeRecorder`, It we create it like this, it will discard the events.